### ISSUE TITLE
False positives on Rails/DynamicFindBy

### ISSUE THREAD

**Issue by jcoyne** [bbatsov/rubocop#3758]
Any method named `find_by_*` is triggering `Rails/DynamicFindBy` errors.  This is occurring even if the methods are statically defined.

For example any call to this method triggers it:
https://github.com/projecthydra/hydra-head/blob/f63dcc2592fbef6a648ae21e85433db7c2a7749f/hydra-access-controls/lib/hydra/user.rb#L20

**Comment by pocke** [comment]
Hi, thanks for your feedback!
I also recognize the problem.
We can use whitelist for the problem.

e.g.

``` yaml
# In .rubocop.yml
Rails/DynamicFindBy:
  Whitelist:
    - find_by_user_key
```

https://github.com/bbatsov/rubocop/issues/3455#issuecomment-253237902

**Comment by bbatsov** [comment]
That's the reason the whitelist exists. There's nothing else we can do (apart from disabling the cop).
